File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let result = 0;
    
    function getDivisor(n) {
        let count = 0;
        const sqrt = Math.sqrt(n);
        
        for (let i = 1; i <= sqrt; i++) {
            if (n % i === 0) {
                count += sqrt === i ? 1 : 2;
            }
        }
        
        return count;
    }
    
    for (let i = 1; i <= number; i++) {
        const divisor = getDivisor(i);
        const attack = divisor > limit ? power : divisor;
        
        result += attack;
    }
    
    return result;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
import math

def solution(number, limit, power):
    D = [0]
    
    for x in range(1, number+1):
        count = 0
        
        for i in range(1, int(math.sqrt(x))+1):
            if x % i == 0:
                count += 1 if i * i == x else 2
                
        D.append(count)
        
    P = [D[x] if D[x] <= limit else power for x in range(1, number+1)]
    return sum(P)
    
    
            
